lib/internal/streams/pipeline.js

```diff
@@ -125,7 +126,9 @@ async function pump(iterable, writable, finish) {
     }
     writable.end();
   } catch (err) {
-    finish(err);
+    error = err;
```

Comment:
This is a little awkward but we need to ensure finish is always called and only once.